logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/call/usecase/IsEligibleToStartCallUseCase.kt

```diff
@@ -0,0 +1,83 @@
+package com.wire.kalium.logic.feature.call.usecase
+
+import com.wire.kalium.logic.configuration.UserConfigRepository
+import com.wire.kalium.logic.data.call.CallRepository
+import com.wire.kalium.logic.data.conversation.Conversation
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.functional.fold
+
+/**
+ * @return the [Boolean] for if the user's team has conference calling enabled in its plan.
+ */
+interface IsEligibleToStartCallUseCase {
+    suspend operator fun invoke(conversationId: ConversationId, conversationType: Conversation.Type): ConferenceCallingResult
+}
+
+internal class IsEligibleToStartCallUseCaseImpl(
+    private val userConfigRepository: UserConfigRepository,
+    private val callRepository: CallRepository
+) : IsEligibleToStartCallUseCase {
+
+    override suspend fun invoke(conversationId: ConversationId, conversationType: Conversation.Type): ConferenceCallingResult {
+        val establishedCallConversationId = callRepository.establishedCallConversationId()
+
+        val canStartCall = (conversationType == Conversation.Type.ONE_ON_ONE ||
+                (conversationType == Conversation.Type.GROUP && isConferenceCallingEnabled()))
+
+        return establishedCallConversationId?.let {
+            callIsEstablished(it, conversationId, canStartCall)
+        } ?: run {
+            if (canStartCall) ConferenceCallingResult.Enabled else ConferenceCallingResult.Disabled.Unavailable
+        }
+    }
+
+    private fun callIsEstablished(
+        establishedCallConversationId: ConversationId,
+        conversationId: ConversationId,
+        canStartCall: Boolean
+    ): ConferenceCallingResult = when {
+        establishedCallConversationId == conversationId -> ConferenceCallingResult.Disabled.Established
+        canStartCall -> ConferenceCallingResult.Disabled.OngoingCall
+        !canStartCall -> ConferenceCallingResult.Disabled.Unavailable
+        else -> ConferenceCallingResult.Disabled.Unavailable
+    }
+
+    private fun isConferenceCallingEnabled(): Boolean = userConfigRepository
+        .isConferenceCallingEnabled()
+        .fold({
+            DEFAULT_CONFERENCE_CALLING_ENABLED_VALUE
+        }, {
+            it
+        })
+
+    private companion object {
+        const val DEFAULT_CONFERENCE_CALLING_ENABLED_VALUE = false
```

Comment:
Should be true as a default value ? if for some reason we fail to retrieve the feature flag, this feature will be disabled for all user

Reply:
I left it as a `false` value due to the comment made by `johnxnguyen` in [this PR](https://github.com/wireapp/wire-android/pull/3449)